src/core/operations/RAKE.mjs

```diff
@@ -101,22 +101,17 @@ class RAKE extends Operation {
         phrases = phrases.filter(subArray => subArray.length > 0);
 
         // Remove duplicate phrases
-        const uniquePhrases = [...new Set(phrases.map(function (phrase) {
-            return phrase.join(" ");
-        }))];
-        phrases = uniquePhrases.map(function (phrase) {
-            return phrase.split(" ");
-        });
-
+        phrases = phrases.unique();
+        
         // Generate word_degree_matrix and populate
-        const wordDegreeMatrix = Array.from(Array(tokens.length), _ => Array(tokens.length).fill(0));
-        phrases.forEach(function (phrase) {
-            phrase.forEach(function (word1) {
-                phrase.forEach(function (word2) {
-                    wordDegreeMatrix[tokens.indexOf(word1)][tokens.indexOf(word2)]++;
-                });
-            });
-        });
+        const wordDegreeMatrix = Array(tokens.length).fill().map(() => Array(tokens.length).fill(0));
+        for (let p=0; p < phrases.length; p++) {
```

Comment:
You can improve the readability of this with:

```
for (let phrase of phrases) {
  for (let word1 of phrase) {
    for (let word2 of phrase) {
      wordDegreeMatrix[tokens.indexOf(word1)][tokens.indexOf(word2)]++;
    }
  }
}
```

Reply:
Sorry have only just seen this now. Ill make the changes.